src/commands/pickFuncProcess.ts

```diff
@@ -27,10 +27,19 @@ export async function pickFuncProcess(actionContext: IActionContext): Promise<st
         throw new Error(localize('noFuncTask', 'Failed to find task with label "{0}".', funcHostTaskLabel));
     }
 
-    const pid: string = await startFuncTask(funcTask, actionContext);
+    const settingKey: string = 'pickProcessTimeout';
+    const settingValue: number | undefined = getFuncExtensionSetting<number>(settingKey);
+    const timeoutInSeconds: number = Number(settingValue);
+    if (isNaN(timeoutInSeconds)) {
+        throw new Error(localize('invalidSettingValue', 'The setting "{0}" must be a number, but instead found "{1}".', settingKey, settingValue));
+    }
+    actionContext.properties.timeoutInSeconds = timeoutInSeconds.toString();
+    const timeoutError: Error = new Error(localize('failedToFindFuncHost', 'Failed to detect running Functions host within "{0}" seconds. You may want to adjust the "{1}" setting.', timeoutInSeconds, `${extensionPrefix}.${settingKey}`));
+
+    const pid: string = await startFuncTask(funcTask, timeoutInSeconds, timeoutError);
     // On Mac/Linux we can leverage the pid of the task directly.
     // On Windows, the pid of the task corresponds to the parent PowerShell process and we have to drill down to get the actual func process
-    return isWindows ? await getInnermostWindowsPid(pid) : pid;
+    return isWindows ? await getInnermostWindowsPid(pid, timeoutInSeconds, timeoutError) : pid;
```

Comment:
nit: Is there any reason to do `await startFuncTask()` before returning?  Seems like you could just do:
`return isWindows ? await getInnermostWindowsPid(pid, timeoutInSeconds, timeoutError) : await startFuncTask(funcTask, timeoutInSeconds, timeoutError);`

Reply:
startFuncTask does a few things, including _starting the task_ (as the name describes) and returning the pid. We need that original pid to find the _inner_ pid